pages/signal-files/[id].js

```diff
@@ -0,0 +1,164 @@
+  attribution = null,
+  signalFile,
+  fileSet,
+  documents,
+  derivedFrom,
+  referenceFiles,
+  isJson,
+}) {
+  return (
+    <>
+      <Breadcrumbs />
+      <EditableItem item={signalFile}>
+        <PagePreamble />
+        <JsonDisplay item={signalFile} isJsonFormat={isJson}>
+          <DataPanel>
+            <DataArea>
+              <FileDataItems
+                file={signalFile}
+                fileSet={fileSet}
+                derivedFrom={derivedFrom}
+              >
+                {referenceFiles.length > 0 && (
+                  <>
+                    <DataItemLabel>Reference Files</DataItemLabel>
+                    <DataItemValue>
+                      <SeparatedList>
+                        {referenceFiles.map((file) => (
+                          <Link href={file["@id"]} key={file.uuid}>
+                            {file.accession}
+                          </Link>
+                        ))}
+                      </SeparatedList>
+                    </DataItemValue>
+                  </>
+                )}
+                <>
+                  <DataItemLabel>Strand Specificity</DataItemLabel>
+                  <DataItemValue>{signalFile.strand_specificity}</DataItemValue>
+                </>
+                {"filtered" in signalFile && (
+                  <>
+                    <DataItemLabel>Filtered</DataItemLabel>
+                    <DataItemValue>
+                      {signalFile.filtered ? "Yes" : "No"}
+                    </DataItemValue>
+                  </>
+                )}
+                {"normalized" in signalFile && (
+                  <>
+                    <DataItemLabel>Normalized</DataItemLabel>
+                    <DataItemValue>
+                      {signalFile.normalized ? "Yes" : "No"}
+                    </DataItemValue>
+                  </>
+                )}
+                {signalFile.start_view_position && (
+                  <>
+                    <DataItemLabel>Start View Position</DataItemLabel>
+                    <DataItemValue>
+                      {signalFile.start_view_position}
+                    </DataItemValue>
+                  </>
+                )}
+              </FileDataItems>
+            </DataArea>
+          </DataPanel>
+          {documents.length > 0 && (
+            <>
+              <DataAreaTitle>Documents</DataAreaTitle>
+              <DocumentTable documents={documents} />
+            </>
+          )}
+          <Attribution attribution={attribution} />
+        </JsonDisplay>
+      </EditableItem>
+    </>
+  );
+}
+
+SignalFile.propTypes = {
+  // SignalFile object to display
+  signalFile: PropTypes.object.isRequired,
+  // File set that contains this file
+  fileSet: PropTypes.object,
+  // Documents set associate with this file
+  documents: PropTypes.array,
+  // The file is derived from
+  derivedFrom: PropTypes.array,
+  // Attribution for this file
+  attribution: PropTypes.object,
```

Comment:
You can set `documents`, `derivedFrom`, and `attribution` to `.isRequired` since they’ll always be _something_, even an empty array or object. Use a default value for `fileSet` just to keep that pattern going.